scapy/arch/libpcap.py

```diff
@@ -362,20 +422,35 @@ def reload(self):
     class L2pcapListenSocket(_L2libpcapSocket):
         desc = "read packets at layer 2 using libpcap"
 
-        def __init__(self, iface=None, type=ETH_P_ALL, promisc=None, filter=None, monitor=None):  # noqa: E501
-            super(L2pcapListenSocket, self).__init__()
+        def __init__(self,
+                     iface=None,  # type: Optional[_GlobInterfaceType]
+                     type=ETH_P_ALL,  # type: int
+                     promisc=None,  # type: Optional[bool]
+                     filter=None,  # type: Optional[str]
+                     monitor=None,  # type: Optional[bool]
+                     ):
+            # type: (...) -> None
             self.type = type
             self.outs = None
-            self.iface = iface
             if iface is None:
                 iface = conf.iface
-            if promisc is None:
-                promisc = conf.sniff_promisc
-            self.promisc = promisc
-            self.ins = open_pcap(iface, MTU, self.promisc, 100,
-                                 monitor=monitor)
+            else:
+                self.iface = iface
```

Comment:
If `iface` is `None`, `self.iface` will not be assigned. Is that on purpose?

Reply:
Hmm, good catch